packages/cli/test/functional/test_site_algolia_plugin/expected/index.html

```diff
@@ -81,16 +81,16 @@
     </panel>
     <p><strong>Popover content should have algolia-no-index class</strong></p>
     <popover effect="fade" header="Title" placement="top">
-      <div slot="content" class="algolia-no-index">Content should have `algolia-no-index` class</div>
+      <div slot="content">Content should have `algolia-no-index` class</div>
       <button class="btn btn-secondary">Trigger should not have `algolia-no-index` class</button>
     </popover>
     <popover effect="fade" header="Title" content="Content as attribute does not require `algolia-no-index` class" placement="top">
       <button class="btn btn-secondary">Trigger should not have `algolia-no-index` class</button></popover>
     <p><strong>Question hint and answer should have algolia-no-index class</strong></p>
     <question>
       Question should not have `algolia-no-index` class
-      <div slot="hint" class="algolia-no-index">Hint should have `algolia-no-index` class</div>
-      <div slot="answer" class="algolia-no-index">Answer should have `algolia-no-index` class</div>
+      <div slot="hint">Hint should have `algolia-no-index` class</div>
+      <div slot="answer">Answer should have `algolia-no-index` class</div>
```

Comment:
Hmm, zeyu, I think this shouldn't be happening. The deprecated slot syntax should be transformed into the shorthand syntax and the node should be shifted one layer deeper. 

I'm currently looking into this to find out why it happens. I discovered that if I remove the <dropdown> from above, the deprecated slot syntax would be transformed properly. 

But I'm still investigating this :O

Reply:
For this problem, I realised that it's `insertTemporaryStyles` that is causing the weird behaviour (any element that has slot after dropdown is not transformed properly). We can see many nodes of `<div class="temp-dropdown temp-dropdown-placeholder"></div>` being created before the `<dropdown>`. Can I check why is `insertTemporaryStyles` needed? 

Also, I just realised another problem for adding the `algolia-no-index` class. Currently, what we have for the selectors are 
`
    'popover div[\\#content]',
    'question div[\\#hint]',
    'question div[\\#answer]',
`

This would work for the following code:
`
<popover>
   <div #content>test</div>
</popover>
`

However, our processing actually shifts the node one layer deeper. So we would actually have the following instead:
`
<popover>
   <template #content>
      <div>test</div>
   </template>
</popover>
`

This means that our selectors would not work, and `algolia-no-index` class won't be added successfully. To overcome this, I think we may need to have the following instead:
`
    'popover template[\\#content]',
    'question template[\\#hint]',
    'question template[\\#answer]',
`

But I just wish to clarify if having `algolia-no-index` class on template html element would break anything? :O

Reply:
> For this problem, I realised that it's `insertTemporaryStyles` that is causing the weird behaviour (any element that has slot after dropdown is not transformed properly). We can see many nodes of `<div class="temp-dropdown temp-dropdown-placeholder"></div>` being created before the `<dropdown>`.

I think we may be having a bug in the expected html file where we have many duplicated `<div class="temp-dropdown temp-dropdown-placeholder"></div>` generated above of `<dropdown>` to combat FOUC. 

From my debugging, I discovered that the problem stems from the function `insertTemporaryStyles` where we add the dropdown placeholders before the dropdown node itself:
```
    $dropdown.before(placeholder);
    $dropdown.prev().addClass(attributes.class).addClass(TEMP_DROPDOWN_PLACEHOLDER_CLASS);
```

I think this somehow messes up the processing of the nodes. All the nodes (e.g. popover, modal) that follows the dropdown would not get processed. I tested it out by checking if any nodes that are of the name "popover", "modal", "panel" are processed after the dropdown but it doesn't seem so. Instead, we are processing the "dropdown" node multiple times, in place of the rest of the nodes (e.g. "popover", "modal", "panel") for the test. 

I think it can be fixed by adding the placeholder after instead. Here's the modified code:
```
    $dropdown.after(placeholder);
    $dropdown.next().addClass(attributes.class).addClass(TEMP_DROPDOWN_PLACEHOLDER_CLASS);
```

I think the above should fix the problem and we can see that in the expected html file, the duplicated temporary dropdown placeholder are gone and we are left with only one of it (which should be the expected behaviour, correct me if I'm wrong).

Reply:
> Otherwise, we can actually use a more specific css selector to place the algolia-no-index class on the div instead.

Not exactly sure how algolia does its indexing. To be safe, let's go with this instead since `template`s are erased by vue after compilation.

> I think we may be having a bug in the expected html file where we have many duplicated <div class="temp-dropdown temp-dropdown-placeholder"></div> generated above of <dropdown> to combat FOUC.

Thanks for investigating this! : )

> I think this somehow messes up the processing of the nodes. All the nodes (e.g. popover, modal) that follows the dropdown would not get processed.

To confirm -- `.after()` would result in a 'normal / fixed' processing order?

If so let's do that then, preferably in a separate PR

Reply:
> To confirm -- `.after()` would result in a 'normal / fixed' processing order? (as a 'sibling' operation seems to be causing this)
> 
> If so let's do that then, preferably in a separate PR

Can I clarify about "normal / fixed" processing order? :O 

Suppose we have: 
```
<modal></modal>
<popover></popover>
<panel></panel>
```

You mean if it would be processed in the sequential order, right? Like `modal` -> `popover` -> `panel`?

If so, yup, .after() would give us the correct sequential processing order.

Reply:
e.g.

```
// suppose these are the last 3 elements at the same level
<dropdown>
<div class="temp-dropdown-placeholder></div> // gets inserted here with after()
<panel> // this actually gets processed
```

Reply:
Yup, that's the behaviour I'm seeing currently :-)